### ISSUE TITLE
<code> tags are not respected and can break edit mode

### ISSUE THREAD

**Issue by atavernor** [yourhead/s3#491]
Add the following to an HTML or a Text stack
`<code>
<div style="background:blue;height:300px;width:100%/>
</code>`
1. The `<code>` tags are not respected and the div will preview as per the inline style.
2. More seriously dependent on the content of the `<code>` tag, edit mode can get broken and prevent further stacks being dropped into place. 

This was noticed by a user who was trying to add some `<head>` and `<style>` tags within a `<code> </code>` pair.

The effect appears to be the same in all themes.

**Comment by yourhead** [comment]
So...  a number of thoughts here.

tl;dr:  i think what you're looking for is the `<xmp>` tag -- which forces the page not to render HTML.  the code tag does not actually have this effect.

e.g.: if we enter the following in an html document and then open the document in Safari and Chrome:

```
<pre><span>neat</span></pre>
<code><span>neat</span></code>
<xmp><span>neat</span></xmp>
```

the output will be:

```
neat
neat
<span>neat</span>
```

i see this same behavior when i use stacks as when i use safari alone. so i think this is the expected behavior.

on the "breaking of edit mode" -- active HTML entered and rendered into edit mode can indeed break edit mode.  although there are potential ways of "fixing" this now -- there were not in 2009 when Stacks 1 came out and for backwards compatability reasons i continue just to render all HTML into the edit mode.  Garbage In ==> Garbage Out

adding a `<code>` tag does not change this behavior -- code tags do not exclude the html from rendering.  the best way to make 100% sure that code doesn't render is to encode the `<` as a `&lt;` -- but since that's not very feasible for a stack -- the xmp tag is solid and works with just about everything -- except of course other xmp tags.

i should also note that the example included in this bug contains an HTML syntax error which adds a bit crazy to this bug and if that is entered it will surely create a problem cause chaos in edit mode.

**Comment by atavernor** [comment]
May be worth making this known to devs who have stacks out there that in Stacks 2 used `<code>` tags  e.g. Gary (DooBox) where the problem first got reported to me (why me I do not know!)

**Comment by yourhead** [comment]
I do not think this behavior has changed in any way. Or is in any way new or interesting. I'm not sure why it's suddenly cropped up.

**Comment by atavernor** [comment]
It cropped up because the Code type stacks stopped working in stacks 3.

Not sure `<xmp>` is a good idea to use as it is obsolete and the use of `<pre>` and `<code>` is recommended  [here](https://developer.mozilla.org/en/docs/Web/HTML/Element/xmp) in its place.  Round and round in circles .:)

**Comment by yourhead** [comment]
yes. xmp is officially deprecated -- the "real" solution is to encode the HTML entities -- `<pre>` is really just for styling -- it doesn't change any behavior as far as i'm aware.

there is some risk that xmp will be removed in some future version of WebKit although I suspect it would be quite difficult without some practical alternative other than using JS to encode entities on the fly.

**Comment by yourhead** [comment]
> It cropped up because the Code type stacks stopped working in stacks 3.

Sounds like something I should know about, is there a bug related to this filed? Or maybe some details to help me find it?

**Comment by atavernor** [comment]
I will sort some details for you with an example - may be a couple of days

**Comment by yourhead** [comment]
take your time. i'm feeling quite poorly since the kids went back to school and infected me with the plague. i'm only working a few hours each day and sleeping the rest. next week will be better i think.
